plan/new.go

```diff
@@ -52,7 +57,25 @@ func buildTestCases(plan *Plan) []TestCase {
 				Plan:      plan,
 				Client:    client,
 				Arguments: testArgs,
+				Skip:      false,
+			}
+
+			for _, filter := range behavior.Filters {
+				// Each filter should be a complete match
+				matchCnt := 0
+				for key, value := range filter.AxisMatches {
+					if (key == behavior.ClientAxis && value == client) ||
+						testArgs[key] == value {
+						matchCnt++
+					}
+				}
+				if matchCnt == len(filter.AxisMatches) {
+					t.Skip = true
+					t.SkipReason = fmt.Sprintf("Pruned by SKIP_%s", strings.ToUpper(behavior.Name))
```

Comment:
Can I just stick to the current message since it is hard to track what filter failed and I've already updated the implementation twice.